mllib/src/main/scala/org/apache/spark/mllib/pmml/export/PMMLTreeModelUtils.scala

```diff
@@ -0,0 +1,282 @@
+      val field = fieldName.get
+
+      val split = node.split.get
+      val featureType = split.featureType
+
+      featureType match {
+        case FeatureType.Continuous => {
+          val value = split.threshold.toString
+          if (isLeft) {
+            new SimplePredicate(field, SimplePredicate.Operator.LESS_OR_EQUAL)
+              .withValue(value)
+          }
+          else {
+            new SimplePredicate(field, SimplePredicate.Operator.GREATER_THAN)
+              .withValue(value)
+          }
+        }
+        case FeatureType.Categorical => {
+          if (split.categories.length > 1) {
+            if (isLeft) {
+              val predicates: List[Predicate] =
+                for (category <- split.categories)
+                  yield
+                  new SimplePredicate(field, SimplePredicate.Operator.EQUAL)
+                    .withValue(category.toString)
+
+              val compoundPredicate = new CompoundPredicate()
+                .withBooleanOperator(CompoundPredicate.BooleanOperator.OR)
+                .withPredicates(predicates.asJava)
+
+              compoundPredicate
+            } else {
+              new True()
+            }
+          }
+          else {
+            val value = split.categories(0).toString
+
+            if (isLeft) {
+              new SimplePredicate(field, SimplePredicate.Operator.EQUAL)
+                .withValue(value)
+            }
+            else {
+              new True()
+            }
+          }
+        }
+      }
+    }
+    else {
+      new True()
+    }
+
+  }
+
+  /** Create equivalent PMML node for given mlLib node. */
+  private def createNode(mlLibNode: Node): PMMLNode = {
+    val node = new PMMLNode()
+      .withId(mlLibNode.id.toString)
+      .withScore(mlLibNode.predict.predict.toString)
+
+    node
+  }
+
+  /** Get PMML datafield based on the mllib split feature. */
+  private def getDataField(mllibNode: Node, fieldName: FieldName): Option[DataField] = {
+    if (!mllibNode.isLeaf && mllibNode.split.isDefined) {
+      val split = mllibNode.split.get
+      val dataField = new DataField()
+        .withName(fieldName)
+        .withDataType(DataType.fromValue(split.threshold.getClass.getSimpleName.toLowerCase))
+        .withOpType(OpType.fromValue(split.featureType.toString.toLowerCase))
+
+      split.featureType match {
+        case FeatureType.Continuous => dataField.withOpType(OpType.CONTINUOUS)
+        case FeatureType.Categorical => {
+          dataField.withOpType(OpType.CATEGORICAL)
+          val categories = split.categories
+            .map(category => new PMMLValue(category.toString)).asJava
+          dataField.withValues(categories)
+        }
+      }
+
+      Some(dataField)
+    }
+    else {
+      None
+    }
+  }
+
+  /** Get PMML mining function type for given mlLib Algo. */
+  private def getPMMLMiningFunctionType(mlLibAlgo: Algo): MiningFunctionType = {
```

Comment:
There is no need to create a function here, coz it is only a match .. case statement.

You may use this:

```scala
dtModel.algo match {
  case ... =>
  case ... =>
}
```